### PR TITLE
refactor(core): Refactor and re-organize code for easier maintenance

### PR DESCRIPTION
**Pull request by mmastrac**
Part of some work to refactor and decouple the various parts of core.

### CODE REVIEW

File: core/runtime.rs [hunk 1]
```diff
@@ -565,6 +522,43 @@ impl JsRuntime {
     js_runtime
   }
 
+  /// Create a new [`v8::OwnedIsolate`] and its global [`v8::Context`] from optional parameters and snapshot.
+  fn create_isolate(
```
Comment (bartlomieju) [inline core/runtime.rs:526]: I suggest to merge `JsRuntime::setup_isolate` into this method. There's no point in having two separate functions that act similar

**Review by bartlomieju — COMMENTED** [review]
One nitpick otherwise LGTM. This will most likely introduce conflicts with https://github.com/denoland/deno/pull/19040. Could you review that PR too?

File: core/runtime.rs (hunk 1 above)
Comment (mmastrac) [reply core/runtime.rs:525]: Ah, yes that's a great idea. Done.

**Comment by mmastrac** [comment]
> One nitpick otherwise LGTM. This will most likely introduce conflicts with #19040. Could you review that PR too?

I reviewed #19040 and it looks great. I'd actually like to land that one right after this because it unlocks some more cleanup I was thinking of doing.

**Review by bartlomieju — APPROVED** [review]

**Comment by bartlomieju** [comment]
> > One nitpick otherwise LGTM. This will most likely introduce conflicts with #19040. Could you review that PR too?
> 
> I reviewed #19040 and it looks great. I'd actually like to land that one right after this because it unlocks some more cleanup I was thinking of doing.

@mmastrac feel free to do so :)

**Pull request merged**